app/scripts/lib/channels/notifier-mixin.js

```diff
@@ -62,13 +62,17 @@ define(function (require, exports, module) {
         return false;
       }
 
-      for (var notificationName in notifications) {
-        var method = notifications[notificationName];
-        if (_.isString(method)) {
-          method = consumer[method];
-        }
-
-        if (_.isFunction(method)) {
+      for (let notificationName in notifications) {
+        let method = notifications[notificationName];
+        if (_.isString(method) && _.isFunction(consumer[method])) {
+          this.on(notificationName, (...args) => {
+            // The level of indirection is used to allow for
+            // late-binding when using sinon spies & stubs.
+            // Without indirection, the original function is
+            // always called.
+            consumer[method](...args);
+          });
+        } else if (_.isFunction(method)) {
           this.on(notificationName, method.bind(consumer));
```

Comment:
@shane-tomlinson am I missing something or does the problem remain for this `else` branch? Should we fix it for both?